src/main/java/javaslang/control/Applicative.java

```diff
@@ -0,0 +1,17 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _ _____
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  / /  _  \   Javaslang
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/  \__/  /   Copyright 2014-now Daniel Dietrich
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/___\_/  \_//  \__/_____/    Licensed under the Apache License, Version 2.0
+ */
+package javaslang.control;
+
+public interface Applicative<T> extends Apply<T> {
```

Comment:
This is the first shot using my infamous `Kind` type constructor fake. It is a direct translation of Scalaz's Validation Applicative (more or less). The list of errors is handled internally by the Validation, i.e. we only make the type E explicit and most probably there will be a method `getFailures()` or something similar which returns a `List<E>` if `isFailure() == true`.

```java
// trait Apply[F[_]] extends Functor[F]
public interface Applicative<F extends Kind<F, ?>, A> extends Functor<A> {

    // def ap[A,B](fa: => F[A])(f: => F[A => B]): F[B]
    <B> Applicative<F, B> ap(Kind<F, ? extends Function<? super A, ? extends B>> f);

    @Override
    <B> Applicative<F, B> map(Function<? super A, ? extends B> f);
}

// type constructor: TYPE<T> is split in TYPE<?> and T
interface Kind<TYPE extends Kind<TYPE, ?>, T> {
}

interface Validation<E, A> extends Kind<Validation<E, ?>, A> {

    // def ap[EE >: E, B](x: => Validation[EE, A => B])(implicit E: Semigroup[EE]): Validation[EE, B]
    <B> Validation<E, B> ap(Validation<E, ? extends Function<? super A, ? extends B>> f);

    <B> Validation<E, B> map(Function<? super A, ? extends B> f);

    // implicit def ValidationApplicative[L: Semigroup]: Applicative[Validation[L, ?]]
    final class ValidationApplicative<E, A> implements Applicative<Validation<E, ?>, A> {

        final Validation<E, A> fa;

        public ValidationApplicative(Validation<E, A> fa) {
            this.fa = fa;
        }

        // def ap[A, B](fa: => Validation[L, A])(f: => Validation[L, A => B]) = fa ap f
        @SuppressWarnings("unchecked")
        @Override
        public <B> ValidationApplicative<E, B> ap(Kind<Validation<E, ?>, ? extends Function<? super A, ? extends B>> f) {
            return new ValidationApplicative<>(fa.ap((Validation<E, ? extends Function<? super A, ? extends B>>) (Object) f));
        }

        // override def map[A, B](fa: Validation[L, A])(f: A => B) = fa map f
        @Override
        public <B> Applicative<Validation<E, ?>, B> map(Function<? super A, ? extends B> f) {
            return new ValidationApplicative<>(fa.map(f));
        }
    }
}
```

Next we could get rid of the class ValidationApplicative because Java has no implicits. However, as you already said, I don't if it is worth to make all this effort just to mark Validation as Applicative.